components/url/lib.rs

```diff
@@ -25,6 +24,29 @@ use url::{Position, Url};
 
 pub use crate::origin::{ImmutableOrigin, MutableOrigin, OpaqueOrigin};
 
+#[derive(Debug)]
+pub enum UrlError {
+    SetUsername,
+    SetIpHost,
+    SetPassword,
+    ToFilePath,
+    FromFilePath,
+}
+
+impl fmt::Display for UrlError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            UrlError::SetUsername => write!(f, "Error setting username"),
+            UrlError::SetIpHost => write!(f, "Error setting IP host"),
+            UrlError::SetPassword => write!(f, "Error setting password"),
+            UrlError::ToFilePath => write!(f, "Error converting to file path"),
+            UrlError::FromFilePath => write!(f, "Error converting from file path"),
+        }
+    }
+}
```

Comment:
I think we can just remove this entirely as the `Debug` implementation should be enough.

Reply:
I think `Display` is there because `std::error::Error` [requires it](https://doc.rust-lang.org/std/error/trait.Error.html), but that can probably be removed as well. Other errors in servo such as `WebGLError` or `TimeRangesError` don't implement `Error` or `Display`, just `Debug`.

Reply:
Makes sense. Let's remove both of these.